azurerm/internal/services/loganalytics/log_analytics_data_export_resource.go

```diff
@@ -0,0 +1,199 @@
+	if err != nil {
+		return fmt.Errorf("%v", err)
+	}
+
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, workspace.Name, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for present of existing Log Analytics Data Export Rule %q (Resource Group %q / workspaceName %q): %+v", name, resourceGroup, workspace.Name, err)
+			}
+		}
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_log_analytics_data_export_rule", *existing.ID)
+		}
+	}
+
+	destination := &operationalinsights.Destination{
+		ResourceID: utils.String(d.Get("destination_resource_id").(string)),
+	}
+
+	parameters := operationalinsights.DataExport{
+		DataExportProperties: &operationalinsights.DataExportProperties{
+			Destination: destination,
+			TableNames:  utils.ExpandStringSlice(d.Get("table_names").(*schema.Set).List()),
+			Enable:      utils.Bool(d.Get("enabled").(bool)),
+		},
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resourceGroup, workspace.Name, name, parameters); err != nil {
+		return fmt.Errorf("creating/updating Log Analytics Data Export Rule %q (Resource Group %q / workspaceName %q): %+v", name, resourceGroup, workspace.Name, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, workspace.Name, name)
+	if err != nil {
+		return fmt.Errorf("retrieving Log Analytics Data Export Rule %q (Resource Group %q / workspaceName %q): %+v", name, resourceGroup, workspace.Name, err)
+	}
+
+	if resp.ID == nil || *resp.ID == "" {
+		return fmt.Errorf("empty or nil ID returned for Log Analytics Data Export Rule %q (Resource Group %q / workspaceName %q) ID", name, resourceGroup, workspace.Name)
+	}
+
+	d.SetId(*resp.ID)
+	return resourceArmOperationalinsightsDataExportRead(d, meta)
+}
+
+func resourceArmOperationalinsightsDataExportRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).LogAnalytics.DataExportClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.LogAnalyticsDataExportID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.WorkspaceName, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Log Analytics %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("retrieving Log Analytics Data Export Rule %q (Resource Group %q / workspaceName %q): %+v", id.Name, id.ResourceGroup, id.WorkspaceName, err)
+	}
+	d.Set("name", id.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("workspace_resource_id", id.WorkspaceID)
+	if props := resp.DataExportProperties; props != nil {
+		d.Set("export_rule_id", props.DataExportID)
+		d.Set("destination_resource_id", flattenArmDataExportDestination(props.Destination))
+		d.Set("enabled", props.Enable)
+		if tableNames := props.TableNames; tableNames != nil {
```

Comment:
i think the utility function checks for this?

Reply:
Yep, just looked... if it's `nil` it returns an empty slice. 🙂